### PR TITLE
docs: document the model contract and fix the API reference

### LINKED ISSUE #1049: [Task]: Model Contract - Documentation, Notebooks & Examples Update
### Summary

Update all documentation, Jupyter notebooks, and example scripts to use the new model contract API (`Sample` numpy-only, `Prediction` dataclass, new `fit()`/`predict()` signatures). This is the final issue — only picked up after Issues 1–5 are merged and the new API is stable.

#### 1. Jupyter Notebooks (`library/examples/`)

All notebooks use the old API (predict returns `list[dict[str, torch.Tensor]]`, Sample accepts torch tensors, etc.).

| Notebook | Key changes needed |
|----------|--------------------|
| `dinov3_zero_shot_classification.ipynb` | Update predict output handling (dict → Prediction), Sample construction |
| `efficientsam3_example.ipynb` | Same + EfficientSAM3 constructor changes |
| `matcher_openvino_export.ipynb` | Update export flow, Matcher constructor, predict output |
| `postprocessing_demo.ipynb` | PostProcessor is no longer nn.Module, update pipeline construction |
| `sam3_aerial_maritime_example.ipynb` | SAM3 constructor, predict output, Sample construction |
| `sam3_canvas_example.ipynb` | Canvas mode Sample construction, predict output |
| `sam3_openvino.ipynb` | SAM3OpenVINO constructor (now OpenVINOModel), predict output |

**Common changes across all notebooks:**
- `model.predict(...)` now returns `list[Prediction]` not `list[dict[str, torch.Tensor]]`
- Access results via `pred.masks`, `pred.scores`, `pred.label_names` instead of `result["pred_masks"]`, etc.
- `Sample(image=numpy_array, ...)` — no more torch tensors in Sample
- Remove any `.cpu().numpy()` calls on results (already numpy)
- Update imports: add `Prediction`, `TensorSample` if used
- Remove `Batch` usage — pass `Sample` or `list[Sample]` directly

#### 2. Example Scripts (`library/examples/`)

| Script | Changes |
|--------|---------|
| `gradio_postprocessing.py` | PostProcessor pipeline construction (no nn.Module), predict output handling |
| `sam3_gradio.py` | SAM3 predict output → Prediction, Sample numpy construction |

#### 3. Library Documentation (`library/docs/`)

| File | Changes |
|------|---------|
| `01-introduction.md` | Update API overview if it shows predict signature |
| `02-quick-start.md` | Update code examples (Sample, predict, output handling) |
| `tutorials/` | Update any tutorial showing model usage |
| `how-to-guides/` | Update guides referencing old API |
| `concepts/` | Update architecture description if it mentions nn.Module base |

#### 4. Sphinx API Docs (`library/sphinx/`). ## - This could be a different PR along with application side changes. 

| File | Changes |
|------|---------|
| `conf.py` | May need module path updates if new files added |
| `instantlearn_module.rst` | Add new modules (torch_base, openvino_base, prediction, model_card) |

#### 5. Application Docs (`application/docs/`) — if referencing library types ## - This could be a different PR along with application side changes. 

| File | Changes |
|------|---------|
| `03-use-instant-learn/02-prompt-model.md` | Update if it shows library predict calls |
| `03-use-instant-learn/03-inference.md` | Update output format description |
| `04-concepts/01-architecture.md` | Update if it describes Model(nn.Module) hierarchy |

### Migration patterns

**Before (old API):**
```python
from instantlearn.models import SAM3
from instantlearn.data import Sample, Batch

model = SAM3(device="cuda")
model.fit(Sample(image=ref_img, masks=ref_masks, categories=["cat"]))

batch = Batch.collate([Sample(image=img1), Sample(image=img2)])
results = model.predict(batch)

for result in results:
    masks = result["pred_masks"].cpu().numpy()
    scores = result["pred_scores"].cpu().numpy()
    labels = result["pred_labels"].cpu().numpy()
```

**After (new API):**
```python
from instantlearn.models import SAM3
from instantlearn.data import Sample, Prediction

model = SAM3(device="cuda")
model.fit(Sample(image=ref_img, masks=ref_masks, categories=["cat"]))

predictions = model.predict([Sample(image=img1), Sample(image=img2)])

for pred in predictions:
    masks = pred.masks        # already numpy (N, H, W)
    scores = pred.scores      # already numpy (N,)
    names = pred.label_names  # already numpy (N,) str
```

### Files to modify

| File/Directory | Action |
|----------------|--------|
| `library/examples/dinov3_zero_shot_classification.ipynb` | Update |
| `library/examples/efficientsam3_example.ipynb` | Update |
| `library/examples/matcher_openvino_export.ipynb` | Update |
| `library/examples/postprocessing_demo.ipynb` | Update |
| `library/examples/sam3_aerial_maritime_example.ipynb` | Update |
| `library/examples/sam3_canvas_example.ipynb` | Update |
| `library/examples/sam3_openvino.ipynb` | Update |
| `library/examples/gradio_postprocessing.py` | Update |
| `library/examples/sam3_gradio.py` | Update |
| `library/docs/02-quick-start.md` | Update |
| `library/docs/tutorials/` | Update relevant files |
| `library/docs/how-to-guides/` | Update relevant files |
| `library/sphinx/instantlearn_module.rst` | Add new modules |
| `application/docs/` | Update if referencing library types |

### Component

Library (instantlearn Framework)

### Acceptance Criteria

- [ ] All notebooks execute without errors (kernel restart + run all)
- [ ] No references to old dict-based prediction format (`result["pred_masks"]`, etc.)
- [ ] No `Batch.collate()` usage in examples — use `list[Sample]` directly
- [ ] No `.cpu().numpy()` on predict outputs (already numpy)
- [ ] All code examples in docs use new `Sample`/`Prediction` API
- [ ] Sphinx docs build cleanly (`make html` in `library/sphinx/`)
- [ ] Quick-start guide works end-to-end with copy-paste

### Implementation Notes

_No response_

### Parent Epic

_No response_

### Dependencies

_No response_

### PR DESCRIPTION
**Pull request by rajeshgangireddy**
## Summary

Completes the library documentation update for the unified model contract and closes #1049.

- Adds concise concepts, architecture, and custom-model guides.
- Updates quick-start, dataset, benchmarking, README, and API-reference documentation.
- Fixes the SAM3 OpenVINO quick-start to use a locally exported IR and explicit text prompt mode.
- Fixes `samples_to_tensors()` to accept every declared `Collatable` input, including image paths.
- Gives `EfficientSAM3` its own model card and adds contract regression tests.
- Removes stale links, stray demo emoji prefixes, and documentation formatting issues.

## Verification

- 589 unit tests pass; 5 skipped.
- Sphinx builds with the existing 113 warnings.
- Ruff remains at the merged branch baseline.

No breaking changes.

### CODE REVIEW

File: library/docs/concepts/01-concepts.md [hunk 1]
```diff
@@ -1,19 +1,118 @@
...
+
+`Batch` wraps a `list[Sample]` and adds batch-level accessors. It is **not**
+required: `fit()` and `predict()` accept a single `Sample`, a `list[Sample]`, or
+a `Batch` interchangeably.
+
+```python
+model.predict(sample)                  # one
+model.predict([sample_a, sample_b])    # several
+model.predict(Batch.collate(samples))  # equivalent
+```
```
[the remark this answers is not in the document]
Comment (rajeshgangireddy) [reply library/docs/concepts/01-concepts.md:95]: Kept this one intentionally: `Batch` isn't deprecated, and this section already leads with "prefer a plain list ... it is the simpler form" right after showing `Batch.collate()` as an equivalent (not preferred) option. Removing it would understate that `Batch` is still fully supported.

File: library/docs/concepts/02-architecture.md [hunk 2]
```diff
@@ -1,15 +1,130 @@
...
+to import torch.
+
+Producing the OpenVINO sibling's input is a one-off, local step:
+
+```python
+SAM3(device="cpu").to_openvino(export_path="./sam3-openvino")
+```
+
+Some models bake the reference into the exported graph, so they require `fit()`
+before `to_openvino()`. `Matcher` and `PerDino` work this way — their exported
```
Comment (LenaRudaya) [inline library/docs/concepts/02-architecture.md:64]:
```suggestion
before `to_openvino()`. `Matcher`, `SoftMatcher` and `PerDino` work this way — their exported
```

File: library/docs/how-to-guides/01-custom-dataset.md [hunk 3]
```diff
@@ -1,15 +1,169 @@
...
+
+`n_shots` controls how many images per category become references. Split the dataset and run a model:
+
+```python
+from instantlearn.models import Matcher
+
+references = dataset.get_reference_dataset()
+targets = dataset.get_target_dataset()
+
+model = Matcher(device="xpu")
```
Comment (LenaRudaya) [inline library/docs/how-to-guides/01-custom-dataset.md:63]: should we provide OpenVINO example for `XPU`?
